compiler/rustc_mir/src/transform/instcombine.rs

```diff
@@ -121,4 +134,36 @@ impl<'tcx, 'a> InstCombineContext<'tcx, 'a> {
             }
         }
     }
+
+    fn find_copynonoverlapping(&self, stmt: &Statement<'tcx>) -> Option<(Local, Local)> {
+        if let StatementKind::CopyNonOverlapping(box CopyNonOverlapping { src, dst, count }) =
+            &stmt.kind
+        {
+            let constant = count.constant()?;
+            if let Some(1) = constant.literal.try_eval_usize(self.tcx, self.param_env) {
+                let src = src.place()?.as_local()?;
+                let dst = dst.place()?.as_local()?;
+
+                return Some((src, dst));
+            }
+        }
+
+        None
+    }
+
+    /// Transform "CopyNonOverlapping(src, dst, 1)" ==> "*dst = *src".
+    fn combine_copynonoverlapping(&self, stmt: &mut Statement<'tcx>) {
```

Comment:
Interesting, why is this part of "instcombine", not "lower_intrinsics"?

Reply:
I would have also expected this to be part of the regular intrinsic lowering pass

Reply:
Yeah, that makes more sense.